src/main/java/org/broadinstitute/dsde/consent/ontology/datause/services/TextTranslationServiceImpl.java

```diff
@@ -88,6 +97,153 @@ public String translatePurpose(String dataUseString) throws IOException {
         return translate(dataUse, TranslateFor.PURPOSE);
     }
 
+    /**
+     * Generate a structured summary of the data use.
+     *
+     * @param dataUse The DataUse
+     * @return DataUseSummary The structured summary
+     */
+    private DataUseSummary translateSummary(DataUse dataUse) {
+        DataUseSummary summary = new DataUseSummary();
+        List<DataUseElement> primary = new ArrayList<>();
+        List<DataUseElement> secondary = new ArrayList<>();
+        if (BooleanUtils.isTrue(dataUse.getGeneralUse())) {
+            primary.add(new DataUseElement("GRU", GRU_POS));
```

Comment:
This is exactly along the lines of what I was thinking for the new structured summary, so thank you. One of the reasons I suggested to store the code and description separately, though, was to avoid manipulating the strings when rendering in the UI. It's not too difficult to do, but for restrictions like `GRU_POS`, the code in brackets would have to be trimmed from the end of the description.

Reply:
That should be easy to do. 👍

Reply:
Done.